### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by paige-kohn**

### CODE REVIEW

File: src/main/java/kohn/maze/Maze.java [hunk 1]
```diff
@@ -1,368 +1,179 @@
...
-                        maze[row][col] = "0";
-                    }
-                }
+    private void generateMaze() {
+        int x, y;
+        for (x = 0; x < rows; x++) {
+            for (y = 0; y < cols; y++) {
+                Cell cell = new Cell(x, y);
+                maze[x][y] = cell;
+                cell.visited = false;
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: `cell.visited` is initialized to false already.

File: src/main/java/kohn/maze/Maze.java [hunk 2]
```diff
@@ -1,368 +1,179 @@
...
-        if (check.equals("N")) {
-            if (current.getCol() - 1 < 0) {
-
-                return checkNeighbors(theMaze, current, directions);
+    private void createPath() {
+        int x = 0;
+        int y = 0;
+        Cell cell = maze[x][y];
+
+        while (!containsUnvisitedCells()) {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: `!containsUnvisitedCells()` is wrong. It's also a double negative.
It should be while there are unvisited cells.

File: src/main/java/kohn/maze/Maze.java [hunk 3]
```diff
@@ -1,368 +1,179 @@
...
+            cell.neighbors.add(maze[cell.getX()][cell.getY() + 1]);
         }
-        return check;
     }
 
+    private Cell getRandomNeighbor(Cell cell) {
+        if (cell.neighbors.size() == 0) {
+            return null;
+        }
+        Random random = new Random();
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: create `random` as a constant, do not recreate it each time you need a random number

File: src/main/java/kohn/maze/Maze.java [hunk 4]
```diff
@@ -1,368 +1,179 @@
...
-                continue;
-            }
-            current = move(theMaze, current, check, visited);
-            visited = visited + 1;
-            location.push(current);
-
-            if (current.getRow() == sz - 1 && current.getCol() == sz - 1) {
-                return theMaze;
-            }
+    private int removeVisitedNeighbors(Cell cell) {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: you're already removing visited neighbors somewhere else...

File: src/main/java/kohn/maze/Maze.java [hunk 5]
```diff
@@ -1,368 +1,179 @@
...
+        for (int i = 0; i < cell.neighbors.size(); i++) {
+            cell.neighbors.removeIf(c -> c.visited);
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: you only need to call this once, not once for each neighboring cell.

File: src/main/java/kohn/maze/Maze.java [hunk 6]
```diff
@@ -1,368 +1,179 @@
...
-                return checkNeighbors(theMaze, current, directions);
+    private void createPath() {
+        int x = 0;
+        int y = 0;
+        Cell cell = maze[x][y];
+
+        while (!containsUnvisitedCells()) {
+            if (removeVisitedNeighbors(cell) != 0) {
+                Cell neighbor = getRandomNeighbor(cell);
+                if(neighbor != null) {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: if `neighbor == null`, then you want to also pop the stack

File: src/main/java/kohn/maze/Maze.java [hunk 7]
```diff
@@ -1,368 +1,179 @@
...
 
-    private String[][] clearPath(String[][] theMaze) {
-        int sz = theMaze.length;
-        for (int i = 0; i < sz; i++) {
-            for (int j = 0; j < sz; j++) {
-                if (theMaze[i][j].equals("#")) {
-                    theMaze[i][j] = " ";
+    private boolean containsUnvisitedCells() {
+        int x, y;
+        for (x = 0; x < rows; x++) {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: `int x = 0` and `int y = 0`

File: src/main/java/kohn/maze/Maze.java (hunk 4 above)
Comment (andrewoid) [reply src/main/java/kohn/maze/Maze.java]: nevermind

**Comment by andrewoid** [comment]
In the future, try to name your branches (and PRs) something more meaningful than just "Homework"

**Comment by paige-kohn** [comment]
Ok, I will. thank you. 



On Sep 9, 2018, at 5:54 PM, Andrew Schwimmer <notifications@github.com> wrote:

In the future, try to name your branches (and PRs) something more meaningful than just "Homework"

File: src/main/java/kohn/maze/Maze.java (hunk 6 above)
Comment (paige-kohn) [reply src/main/java/kohn/maze/Maze.java]: Doesn't the else within the while loop control that already?

File: src/main/java/kohn/earthquake/net/EarthquakeController.java [hunk 8]
```diff
@@ -6,86 +6,61 @@
...
-		viewProvider.get().setEarthquakes(largest);
-
-		largestField.setText(String.valueOf(largest.get(4).getMagnitude()) + ", ");
-	}
-
-
+    private USGSEarthquakeService service;
+
+    private Provider<EarthquakeView> viewProvider;
+    String month = "month";
```
Comment (andrewoid) [inline src/main/java/kohn/earthquake/net/EarthquakeController.java]: These should be constants

File: src/main/java/kohn/earthquake/net/EarthquakeController.java [hunk 9]
```diff
@@ -6,86 +6,61 @@
...
+    String day = "day";
+    String week = "week";
+    String hour = "hour";
+    Disposable disposable;
```
Comment (andrewoid) [inline src/main/java/kohn/earthquake/net/EarthquakeController.java]: private

**Review by andrewoid — APPROVED** [review]

**Pull request merged**